libcontainer/apparmor/apparmor.go

```diff
@@ -24,15 +21,29 @@ func IsEnabled() bool {
 	return false
 }
 
+func setprocattr(tid int, attr, value string) error {
+	path := fmt.Sprintf("/proc/%d/attr/%s", tid, attr)
+
+	f, err := os.OpenFile(path, os.O_WRONLY, 0)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	_, err = fmt.Fprintf(f, "%s", value)
+	return err
+}
+
 // ApplyProfile will apply the profile with the specified name to the process after
 // the next exec.
 func ApplyProfile(name string) error {
 	if name == "" {
 		return nil
 	}
```

Comment:
Should check `!strings.Contains(name, "/")`

Reply:
Why? Profile names can have `/` in them?

Reply:
Would something like the following be sufficient?

```
if strings.Contains(name, "/") {
    return fmt.Errorf("apparmor profile name cannot contain '/'")
}
```

Reply:
At least ".." should be prohibited to prevent potential filesystem contamination

Reply:
np, I misread the code